fenix/app/src/main/java/org/mozilla/fenix/debugsettings/ui/DebugOverlay.kt

```diff
@@ -30,7 +40,25 @@ import org.mozilla.fenix.theme.FirefoxTheme
 @Composable
 fun DebugOverlay() {
     val snackbarState = remember { SnackbarHostState() }
-    val scope = rememberCoroutineScope()
+    val drawerState = rememberDrawerState(initialValue = DrawerValue.Closed)
+    var drawerEnabled by remember { mutableStateOf(false) }
+
+    LaunchedEffect(drawerEnabled) {
+        Log.d("bondbond", "drawerEnabled update - $drawerEnabled")
+        if (drawerEnabled) {
+            drawerState.open()
+        }
+    }
+
+    LaunchedEffect(drawerState) {
+        snapshotFlow { drawerState.currentValue }
```

Comment:
is it possible the above `drawerState.open()` is not completed before we run this block?

Reply:
Good question

All calls to `drawerState.open()` should be completed before we get into the actual meat of the launched effect with `drawerState`. 
1. The snapshotFlow is being derived from `currentValue`, so it is only being triggered when the drawer state changes and has settled. There is actually a separate API `targetValue` which would allow us to listen to changes to what state the drawer is being moved to. If we were _just_ listening and acting upon all changes to drawerState (without a snapshotFlow on the parameter we cared about), then, yes, this would have been called preemptively.
2. Thanks to snapshotFlow + `distinctUntilChanged` + the `if`,  we're only listening for state changes where it has changed from `open` to `closed`. 

I'll add a quick polish to the snapshot flow, so now it'll be:
```
snapshotFlow { drawerState.currentValue }
            .distinctUntilChanged()
            .filter { it == DrawerValue.Closed }
            .collect {
                drawerEnabled = false
            }
```
With the filter, it's a little cleaner that we're only wanting state emissions of drawer state going from open to closed.

Reply:
Thanks @MozillaNoah, appreciate the info here! Looks great 🙂 

I think you can simplify:
```koltin
.distinctUntilChanged()
.filter { it == DrawerValue.Closed }
```
to
```kotlin
.distinctUntilChangedBy { it == DrawerValue.Closed }
```